### PR TITLE
Change query range response structure

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
This PR is for changing the response structure of query_range table function in Prometheus Connector.

BWC failures are unrelated. WIll get fixed with the below PRs
https://github.com/opensearch-project/sql/pull/1876
https://github.com/opensearch-project/sql/pull/1868

Request::
PUT http://localhost:9200/_plugins/_ppl
```
{
    "query" : "source = my_prometheus.query_range('prometheus_http_requests_total',1689281439,1689291439,14)"
}
```
Sample Old Response
```
{
    "schema": [
        {
            "name": "@timestamp",
            "type": "timestamp"
        },
        {
            "name": "@value",
            "type": "double"
        },
        {
            "name": "handler",
            "type": "string"
        },
        {
            "name": "code",
            "type": "string"
        },
        {
            "name": "instance",
            "type": "string"
        },
        {
            "name": "__name__",
            "type": "string"
        },
        {
            "name": "job",
            "type": "string"
        }
    ],
    "datarows": [
        [
            "2023-07-17 23:57:07",
            5.0,
            "/api/v1/query_range",
            "200",
            "localhost:9090",
            "prometheus_http_requests_total",
            "prometheus"
        ],
        [
            "2023-07-17 23:49:39",
            1.0,
            "/metrics",
            "200",
            "localhost:9090",
            "prometheus_http_requests_total",
            "prometheus"
        ]
    "total": 34,
    "size": 34
}
```
There is no segregation of data among different timeseries in a metric.

In the new response below...each timeseries of a metric is a row individually.

Response::
```
{
    "schema": [
        {
            "name": "@timestamp",
            "type": "array"
        },
        {
            "name": "@value",
            "type": "array"
        },
        {
            "name": "@labels",
            "type": "struct"
        }
    ],
    "datarows": [
        [
            {
                "handler": "/api/v1/query_range",
                "code": "200",
                "instance": "localhost:9090",
                "__name__": "prometheus_http_requests_total",
                "job": "prometheus"
            },
            [
                "2023-07-13 23:37:15"
            ],
            [
                3.0
            ]
        ],
        [
            {
                "handler": "/metrics",
                "code": "200",
                "instance": "localhost:9090",
                "__name__": "prometheus_http_requests_total",
                "job": "prometheus"
            },
            [
                "2023-07-13 23:33:17",
                "2023-07-13 23:33:31",
                "2023-07-13 23:33:45",
                "2023-07-13 23:33:59",
                "2023-07-13 23:34:13",
                "2023-07-13 23:34:27",
                "2023-07-13 23:34:41",
                "2023-07-13 23:34:55",
                "2023-07-13 23:35:09",
                "2023-07-13 23:35:23",
                "2023-07-13 23:35:37",
                "2023-07-13 23:35:51",
                "2023-07-13 23:36:05",
                "2023-07-13 23:36:19",
                "2023-07-13 23:36:33",
                "2023-07-13 23:36:47",
                "2023-07-13 23:37:01",
                "2023-07-13 23:37:15"
            ],
            [
                1.0,
                2.0,
                3.0,
                4.0,
                5.0,
                6.0,
                7.0,
                7.0,
                8.0,
                9.0,
                10.0,
                11.0,
                12.0,
                13.0,
                14.0,
                15.0,
                16.0,
                17.0
            ]
        ]
    ],
    "total": 2,
    "size": 2
}
```
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by pjfitzgibbons — DISMISSED** [review]

**Comment by dai-chen** [comment]
There is only IT for explain? Could you add to your PR description the response format before/after your changes?

**Comment by vamsi-amazon** [comment]
> There is only IT for explain? Could you add to your PR description the response format before/after your changes?
@dai-chen  Added old response. Its hard to verify the row response..but will add IT for atleast checking schema output.

File: prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusMetricTable.java [hunk 1]
```diff
@@ -97,10 +97,6 @@ public Map<String, ExprType> getFieldTypes() {
   public PhysicalPlan implement(LogicalPlan plan) {
     PrometheusMetricScan metricScan =
         new PrometheusMetricScan(prometheusClient);
-    if (prometheusQueryRequest != null) {
-      metricScan.setRequest(prometheusQueryRequest);
```
Comment (dai-chen) [inline prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusMetricTable.java:101]: I see `request` field is still in use in metric scan. So where we set it now?

File: prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java [hunk 2]
```diff
@@ -94,14 +94,12 @@ public PhysicalPlan visitIndexAggregation(PrometheusLogicalMetricAgg node,
   public PhysicalPlan visitRelation(LogicalRelation node,
                                     PrometheusMetricScan context) {
     PrometheusMetricTable prometheusMetricTable = (PrometheusMetricTable) node.getTable();
-    if (prometheusMetricTable.getMetricName() != null) {
```
Comment (dai-chen) [inline prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java:97]: Just my feel that it's a little hard to follow why several `if` check removed in this PR. it's fine if it's covered in UT.

Comment (vamsi-amazon) [reply prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java:97]: In earlier implementation, query_range function implementation got coupled with normal PPL queries source = my_prometheus.proemtheus_http_requests_total implementation. This PR at least decouples few of the things.

Need to refactor the existing implementation. I tried to refactor in this PR, the changes are becoming huge. Will cover them in another PR.

File: prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusMetricTable.java (hunk 1 above)
Comment (vamsi-amazon) [reply prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusMetricTable.java:101]: It is set in https://github.com/opensearch-project/sql/blob/main/prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java#L109 here

Implementation of query_range and ppl.prometheus_http_requests_total got coupled with weird logic. Need to refactor in another PR.

File: prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java (hunk 2 above)
Comment (vamsi-amazon) [reply prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java:97]: With the new implementation. Visit Relation will always have metricName.

File: prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusMetricTable.java (hunk 1 above)
Comment (vamsi-amazon) [reply prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusMetricTable.java:101]: query_range...we directly set the prometheusQueryRequest.
In case of `source = my_prometheus.prometheus_http_requests_total` we set these parameters in PrometheusDefaultImplementator class.

File: prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java (hunk 2 above)
Comment (dai-chen) [reply prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java:97]: I see. Please add TODO comment or put all refactor items in Github issue. Thanks!

**Review by dai-chen — APPROVED** [review]

File: prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java (hunk 2 above)
Comment (vamsi-amazon) [reply prometheus/src/main/java/org/opensearch/sql/prometheus/storage/implementor/PrometheusDefaultImplementor.java:97]: https://github.com/opensearch-project/sql/issues/1884

**Review by rupal-bq — APPROVED** [review]
